### ISSUE TITLE
This package is GREAT!!

### ISSUE THREAD

**Issue by loukesio** [sherrillmix/levenR#1]
Dear sherrillmix, 

I have found levenR so cool and so useful!
You should definitely put it in CRAN.
excellent work

**Comment by sherrillmix** [comment]
Thanks for the encouragement loukesio.  I probably should polish it up a bit and submit. On the other hand, I use it for convenience but I had always figured the biostrings package had most use cases covered more completely. Is it filling a gap in biostrings for you?

**Comment by loukesio** [comment]
Yes definitely!  It helps me in so many different ways! 
I cluster my strings, I can align them and have a short look! 
Go for it! 
Great job

**Comment by loukesio** [comment]
Dear sherrillmix, 

At the moment I am using levenR to measure the distance and cluster 2*10^6 sequences which are 20bp long. 
Is there any advice that you could give to speed up the process?

We have a HPC which I can use! Have you ever used that many sequences? 

thank you for your time, 
any hint or advice is appreciated

**Comment by sherrillmix** [comment]
Hmm that might be a bit of a stress test. Especially if that's 2M vs 2M? The output alone would be a 2M x 2M matrix so several Terabytes of RAM.

What's the goal? If everything is 20bp long perhaps you don't need to worry about indels and Hamming distance would do you?

**Comment by loukesio** [comment]
thank you sherrillmix for your time! To make the difference I really need to calculate the lv distance. 
I have used 
levenR 
Biostrings::stringDist
stringdist::stringdist
tidystringdist 
and I can calculate nicely up to 20000 seqs. When I go above 100.000 things go tough hahahah
thank you for taking the time

**Comment by sherrillmix** [comment]
Yeah anything more than ~40000 x ~40000 is probably going to start requiring some custom programming because you end up with more entries than a 4 byte integer can provide and R can't really keep track of things at that point. So I think at that scale you might need to dip into something that allows a bit more memory management and/or higher bit integers than R does easily e.g. C/C++.
